src/platform/forms-system/src/js/state/helpers.js

```diff
@@ -495,16 +502,25 @@ export function updateSchemaAndData(schema, uiSchema, formData) {
   // Update the schema with any general updates based on the new data
   newSchema = updateSchemaFromUiSchema(newSchema, uiSchema, formData);
 
-  // Remove any data that’s now hidden in the schema
-  const newData = removeHiddenData(newSchema, formData);
+  if (!preserveHiddenData) {
+    // Remove any data that’s now hidden in the schema
+    const newData = removeHiddenData(newSchema, formData);
 
-  // We need to do this again because array data might have been removed
-  newSchema = updateItemsSchema(newSchema, newData);
+    // We need to do this again because array data might have been removed
+    newSchema = updateItemsSchema(newSchema, newData);
```

Comment:
I'm not sure you'd want to skip this, because a user could still intentionally remove an array row and we'd want to keep the `items` schema in sync.

Reply:
I'm not sure I follow. We call `updateItemsSchema` as the first step in this function. Are you saying we just shouldn't allow not deleting newly hidden data at all?

Reply:
Oh, nevermind, I just missed that line, forgot we were doing this twice.

Reply:
Whew! I thought I was losing it!